packages/components/src/mobile/link-settings/image-link-destinations-screen.native.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * External dependencies
+ */
+import { useNavigation, useRoute } from '@react-navigation/native';
+import { StyleSheet } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Icon, check, chevronRight } from '@wordpress/icons';
+import { blockSettingsScreens } from '@wordpress/block-editor';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import PanelBody from '../../panel/body';
+import BottomSheet from '../bottom-sheet';
+
+const LINK_DESTINATION_NONE = 'none';
+const LINK_DESTINATION_MEDIA = 'media';
+const LINK_DESTINATION_ATTACHMENT = 'attachment';
+const LINK_DESTINATION_CUSTOM = 'custom';
+
+function LinkDestination( {
+	children,
+	currentOption,
+	label,
+	option,
+	onPress,
+	value,
+	valueStyle,
+} ) {
+	const optionIcon = usePreferredColorSchemeStyle(
+		styles.optionIcon,
+		styles.optionIconDark
+	);
+	return (
+		<BottomSheet.Cell
+			icon={ check }
+			iconStyle={ StyleSheet.flatten( [
+				optionIcon,
+				currentOption !== option && styles.unselectedOptionIcon,
+			] ) }
+			label={ label }
+			leftAlign
+			onPress={ onPress }
+			value={ value }
+			valueStyle={ valueStyle }
+			separatorType="leftMargin"
+		>
+			{ children }
+		</BottomSheet.Cell>
+	);
+}
+
+function ImageLinkDestinationsScreen( props ) {
+	const navigation = useNavigation();
+	const route = useRoute();
+	const { url = '' } = props;
+	const { inputValue = url, imageUrl, attachmentPageUrl, linkDestination } =
+		route.params || {};
+
+	const customUrlSet =
+		!! inputValue &&
+		inputValue !== imageUrl &&
+		inputValue !== attachmentPageUrl;
+
+	function goToLinkPicker() {
+		navigation.navigate( blockSettingsScreens.linkPicker, {
+			inputValue: customUrlSet ? inputValue : '',
+		} );
+	}
+
+	const setLinkDestination = ( newLinkDestination ) => () => {
+		let newUrl;
+		switch ( newLinkDestination ) {
+			case LINK_DESTINATION_MEDIA:
+				newUrl = imageUrl;
+				break;
+			case LINK_DESTINATION_ATTACHMENT:
+				newUrl = attachmentPageUrl;
+				break;
+			default:
+				newUrl = '';
+				break;
+		}
+
+		navigation.navigate( blockSettingsScreens.settings, {
+			inputValue: '', // Reset to avoid stale value in link picker input
+			text: undefined, // Reset to avoid stale value in link picker input
+			url: newUrl,
+			linkDestination: newLinkDestination,
+		} );
+	};
+
+	return (
+		<>
+			<BottomSheet.NavBar>
+				<BottomSheet.NavBar.BackButton onPress={ navigation.goBack } />
+				<BottomSheet.NavBar.Heading>
+					{ __( 'Link To' ) }
+				</BottomSheet.NavBar.Heading>
+			</BottomSheet.NavBar>
+			<PanelBody>
+				<LinkDestination
+					currentOption={ linkDestination }
```

Comment:
NP: Is the `currentOption` only needed by the `LinkDestination` component to conditionally render the check mark icon? I wonder if the intent would be more "readable" with an `isSelected` boolean prop.

Something like:

`isSelected={ linkDestination === LINK_DESTINATION_NONE }`

Wdyt?

Reply:
Yes, that is correct. `currentOption` is solely used for the conditional logic for a selected status. I agree refactoring to `isSelected` is an improvement. Addressed this in https://github.com/WordPress/gutenberg/pull/34846/commits/fa8cea959f4a84dd8000993742652f1fca2849a9.